File: src/Python/747.DominantIndex.py

class Solution:
    def dominantIndex(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        maxnum = -1
        secondmaxnum = -1
        maxindex = 0
        for i, num in enumerate(nums):
            if num > maxnum:
                secondmaxnum = maxnum
                maxnum = num
                maxindex = i
            elif num > secondmaxnum:
                secondmaxnum = num
        return maxindex if maxnum >= 2 * secondmaxnum else -1

if __name__ == '__main__':
    solution = Solution()
    print(solution.dominantIndex([3, 6, 1, 0]))
    print(solution.dominantIndex([1, 2, 3, 4]))
else:
    pass

File: src/Java/747.DominantIndex.java

import java.awt.List;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.Comparator;
import java.util.LinkedList;
import java.util.Queue;
import java.util.HashMap;
import java.util.HashSet;

class Solution {
    public int dominantIndex(int[] nums) {
        int maxnum = -1;
        int secondmaxnum = -1;
        int maxindex = 0;
        for (int i = 0; i < nums.length; i++) {
            int num = nums[i];
            if (num > maxnum){
                secondmaxnum = maxnum;
                maxnum = num;
                maxindex = i;
            }
            else if (num > secondmaxnum)
                secondmaxnum = num;
        }
        return maxnum >= 2 * secondmaxnum ? maxindex : -1;
    }
}
